### ISSUE TITLE
Backing out of a theme option returns you to the top of the list.

### ISSUE THREAD

**Issue by ianklatzco** [Simply-Love/Simply-Love-SM5#140]
It should instead take you back to the same item you were on when you entered that item.

I'm feeling a bit of deja vu as I write this, so it's possible I've reported it before, but I'm not sure if it was an in-person conversation or if it ever actually hit the issue tracker.

![h](https://user-images.githubusercontent.com/1487413/48469970-c1200c80-e7b5-11e8-973c-ee3d399646c2.gif)

**Comment by dguzek** [comment]
I very much agree–this would be a helpful addition to theme.

As it stands, all the Operator Menu screens inherit from `ScreenOptions`, which doesn't currently expose any way of setting the current OptionRow from Lua.

Thus, someone would have to either:

   * build new Lua hooks into the engine, or
   * rewrite the Operator Menu in Lua to not rely on the engine's ScreenOptions stuff

**Comment by ianklatzco** [comment]
You can pass information into new screens via the ... syntax iirc. 

Can you pass information to previous screens (guessing via some kind of return, or updating a global).

**Comment by dguzek** [comment]
That's true.  I'm not sure I understand how that helps here, however.  Can you expand on your idea a little when you have time?

**Comment by dguzek** [comment]
How would you simulate keypresses?

**Comment by ianklatzco** [comment]
Whoops, we have that automated on our machine with `xdotool` and I forgot that's not something the engine exposes.

**Comment by ianklatzco** [comment]
I see, so moving to a given option row is tricky. Thought process is below.

---

It seems like rewriting ScreenOptions in Lua is a bit heavyhanded and requires rewriting something that shouldn't need to be rewritten.

On the flip side, I'm not sure how easy it'd be to add to engine. Sounds like just a "ScreenOptionsParentRow" global?

Checking the API, there's a `GetCurrentRowIndex` [function](https://dguzek.github.io/Lua-For-SM5/API/Lua.xml#ScreenOptions) that does exactly that, I think. If stored when entering a child row, then I think all that's left is to figure out how to move to a given row. And therein lies the problem, aha.

**Comment by sjrc6** [comment]
This seems to be resolved now
